lib/modules/platform/gitea/utils.ts

```diff
@@ -7,3 +16,44 @@ export function smartLinks(body: string): string {
 export function trimTrailingApiPath(url: string): string {
   return url?.replace(regEx(/api\/v1\/?$/g), '');
 }
+
+export function getRepoUrl(
+  repo: helper.Repo,
+  gitUrl: GitUrlOption | undefined,
+  endpoint: string
+): string {
+  if (gitUrl === 'ssh') {
+    if (!repo.ssh_url) {
+      throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+    }
+    logger.debug({ url: repo.ssh_url }, `using SSH URL`);
+    return repo.ssh_url;
+  }
+
+  // Find options for current host and determine Git endpoint
+  const opts: HostRuleSearchResult = hostRules.find({
+    hostType: PlatformId.Gitea,
+    url: endpoint,
+  });
+
+  if (gitUrl === 'endpoint' || !repo.clone_url) {
+    if (!repo.clone_url) {
+      logger.debug('No clone_url found. Falling back to endpoint behaviour.');
+    }
```

Comment:
@rarkins i think we should not fallback here and throw git url error instead, as the old code also failed (with different error) before.

Reply:
I don't really mind. Is your concern about backwards compatibility or that it's the wrong thing to do?

Reply:
i think it's more consistent to throw, as it's done before (because of missing clone url).

we don't need to fallback for backwards compatibility, as we never used the endpoint stuff before.

Reply:
So the suggestion would be to throw a CONFIG_GIT_URL_UNAVAILABLE error, right?

Reply:
yes, like for the ssh case